pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -150,115 +179,133 @@ func (computeCfg *computeServiceConfig) getManagedVnetIDs() map[string]struct{}
-	if snapshot == nil {
-		azurePluginLogger().V(4).Info("Compute service cache snapshot nil", "type", providerType, "account", computeCfg.account)
+	vpcSnapshot := computeCfg.getAccountLevelSnapshot()
+	if vpcSnapshot == nil {
+		azurePluginLogger().Info("Account cache snapshot nil",
+			"type", providerType, "account", computeCfg.accountNamespacedName)
 		return nil
 	}
+
 	vnetPeersCopy := make([][]string, 0)
-	if peers, ok := snapshot.(*computeResourcesCacheSnapshot).vnetPeers[vnetID]; ok {
+	if peers, ok := vpcSnapshot.vnetPeers[vnetID]; ok {
 		vnetPeersCopy = deepcopy.Copy(peers).([][]string)
 	}
 	return vnetPeersCopy
 }
 
-func (computeCfg *computeServiceConfig) getVirtualMachines() ([]*virtualMachineTable, error) {
-	filters, hasFilters := computeCfg.getComputeResourceFilters()
-	if !hasFilters {
-		azurePluginLogger().V(1).Info("Fetching vm resources from cloud skipped",
-			"account", computeCfg.account, "resource-filters", "not-configured")
-		return nil, nil
-	}
-
-	if filters == nil {
-		azurePluginLogger().V(1).Info("Fetching vm resources from cloud",
-			"account", computeCfg.account, "resource-filters", "all(nil)")
+// getSelectorConfigsToPoll finds the list of CES for which making a cloud api call is required.
+func (computeCfg *computeServiceConfig) getSelectorConfigsToPoll() []*selectorConfig {
+	var cesConfigs []*selectorConfig
+	if len(computeCfg.configChangedSelectors) == 0 {
+		// get all CES for cloud api calls.
+		for _, cesConfig := range computeCfg.selectorConfigMap {
+			cesConfigs = append(cesConfigs, cesConfig)
+		}
 	} else {
-		azurePluginLogger().V(1).Info("Fetching vm resources from cloud",
-			"account", computeCfg.account, "resource-filters", "configured")
-	}
-	var subscriptions []*string
-	subscriptions = append(subscriptions, &computeCfg.credentials.SubscriptionID)
-
-	var virtualMachines []*virtualMachineTable
-	for _, filter := range filters {
-		virtualMachineRows, _, err := getVirtualMachineTable(computeCfg.resourceGraphAPIClient, filter, subscriptions)
-		if err != nil {
-			return nil, err
+		for _, namespacedName := range computeCfg.configChangedSelectors {
+			if cesConfig, ok := computeCfg.selectorConfigMap[*namespacedName]; ok {
+				cesConfigs = append(cesConfigs, cesConfig)
+			}
 		}
-		virtualMachines = append(virtualMachines, virtualMachineRows...)
+		computeCfg.configChangedSelectors = nil
 	}
-
-	azurePluginLogger().V(1).Info("Vm instances from cloud",
-		"account", computeCfg.account, "instances", len(virtualMachines))
-
-	return virtualMachines, nil
+	return cesConfigs
 }
 
-func (computeCfg *computeServiceConfig) getComputeResourceFilters() ([]*string, bool) {
-	var allFilters []*string
-
-	computeFilters := computeCfg.computeFilters
-	if len(computeFilters) == 0 {
-		return nil, false
+// TODO: getVirtualMachines should be able to retrieve VM based on VNET ids as well.
+func (computeCfg *computeServiceConfig) getVirtualMachines(updateSnapshot bool) ([]*virtualMachineTable, error) {
```

Comment:
Change this to take selector as input and return virtual machine for that selector

Reply:
yes, changed.